dipy/tracking/streamlinespeed.pyx

```diff
@@ -305,7 +363,7 @@ def set_number_of_points(streamlines, nb_points=3):
             raise ValueError("All streamlines must have at least 2 points.")
 
     # Allocate memory for each modified streamline
-    modified_streamlines = []
+    new_streamlines = []
```

Comment:
@MarcCote shouldn't this be Streamlines rather than []?

Reply:
This part of the code handles the case where a list of ndarrays was provided. In other words, if the user calls `set_number_of_points` on a list of ndarrays, we return a list of modified ndarrays. If the user provides an `ArraySequence` (a.k.a. `dipy.tracking.Streamlines`) object, we return a modified `ArraySequence`. Let me know if that behavior is not what you were expecting.